util/design/lib/OtpMemMap.py

```diff
@@ -165,19 +165,34 @@ def _validate_mmap(config):
 
     offset = 0
     num_part = 0
+    part_index = OrderedDict()
     for part in config["partitions"]:
-        num_part += 1
         _validate_part(part, offset, key_names)
 
+        if part['name'] in part_index:
+            log.error('Partition name {} is not unique'.format(part['name']))
+            exit(1)
```

Comment:
Have you considered using exceptions for this? You could raise a `RuntimeError` in this code (which takes one less statement at each error case) and then catch and do the `log.error`, `exit(1)` dance further up.

Reply:
That's a great suggestion! 
Yes I actually thought about doing that, especially now that the memory map and image generation got more complex (this started out as a simple script, but things got more complex and I had to move this into its own class).
Let me fix this.